### PR TITLE
MNG-150_Create_GET_basket_by_id

### PR DESCRIPTION
**Pull request by mariusz-sm**

### CODE REVIEW

File: src/routes/basket.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Router } from 'express';
+import checkToken from  'express-jwt';
+import { getBasketById, addBasket } from '@entities/Basket/Basket.controller';
+
+const router = Router();
+
+
+router
+    // .get('', getBaskets)
+    .get('/:id', getBasketById)
```
Comment (AleksandraCyp) [inline src/routes/basket.ts]: Basket should be available only to the offerer and the recipient! Add JWT validation, like with POST. Plus I would put it in the 'me' router, not here (it is data about the currently logged user)

I see you added .get('', getBaskets). We will need a endpoint to get all baskets, but again - only of the currently logged user. This for sure should ge the the 'me' router :)

File: src/entities/Basket/Basket.controller.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    try {
+        const basket = await Basket
+            .findById(req.params.id)
+            .populate('createdByUserId', 'name')
+            .populate('targetUserID', 'name')
+        return res.status(OK).json(basket);
+    } catch (error) {
+        return res.status(BAD_REQUEST).send(error.message);
+    }
+}
```
Comment (AleksandraCyp) [inline src/entities/Basket/Basket.controller.ts]: Please add checking if the user has the right to see the basket - he/she has to be either the offerer or the recipient (use JWT)

**Review by AleksandraCyp — CHANGES_REQUESTED** [review]

File: src/entities/Basket/Basket.controller.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+const { BAD_REQUEST, CREATED, OK } = StatusCodes;
+
+export const getBasketById = async (req: Request, res: Response) => {
+    try {
+        const basket = await Basket
+            .findById(req.params.id)
+            .populate('createdByUserId', 'name')
+            .populate('targetUserID', 'name')
+        return res.status(OK).json(basket);
```
Comment (AleksandraCyp) [inline src/entities/Basket/Basket.controller.ts]: We will need also the books - so please populate them too :)

File: src/routes/basket.ts (hunk 1 above)
Comment (mariusz-sm) [reply src/routes/basket.ts]: JWT validation added -> 6a1235915dbc4b3e84ce1c48cace36472841ef52
Get all baskets for logged user -> aed1057373f0038580f2cfb9dee2628a41d0aa4b

File: src/entities/Basket/Basket.controller.ts (hunk 2 above)
Comment (mariusz-sm) [reply src/entities/Basket/Basket.controller.ts]: Could you give a a hint how to do it with JWT?

File: src/entities/Basket/Basket.controller.ts (hunk 3 above)
Comment (mariusz-sm) [reply src/entities/Basket/Basket.controller.ts]: DONE -> 6f7e3908c7bc082288764d24466abc46ce48508f
I need to change schemat to apply 'populate'

File: src/entities/Basket/Basket.controller.ts (hunk 2 above)
Comment (AleksandraCyp) [reply src/entities/Basket/Basket.controller.ts]:
```suggestion
export const getBasketById = async (req: Request, res: Response) => {
  const user= await User.findById(req.user);
  if (!user) return res.status(BAD_REQUEST).send("The request is not sent by logged user");
    try {
        const basket = await Basket
            .findById(req.params.id)
            .populate('createdByUserId', 'name')
            .populate('targetUserID', 'name')
        if (!basket.createdByUserId.equals(user._id) && !basket.targetUserID.equals(user._id)) {
        return.status(FORBIDDEN).send("Basket does not belong to the user"}   
        return res.status(OK).json(basket);
    } catch (error) {
        return res.status(BAD_REQUEST).send(error.message);
    }
}
```
Something like that should work, although of course I am not sure if that's the best way to do it ;)

Comment (mariusz-sm) [reply src/entities/Basket/Basket.controller.ts]: DONE -> 40dbc6cdcaa41a21b521efd1c203871df584c7ca

**Review by AleksandraCyp — APPROVED** [review]

**Pull request merged**